### PR TITLE
Reintegrate the simple single-seller auction after unified refactor

### PR DESCRIPTION
**Pull request by siddharth-sahoo**
# Description

With all existing use-cases migrated to use sequential auction setups, we are left with 2 auction executor scripts in the `ad-tech` service's `src/public/js/ssp` directory -- the newer `run-sequential-ad-auction.js` and the older `run-ad-auction.js`. In addition to using sequential auction setups as the name suggests, the newer `run-sequential-ad-auction.js` is also capable to handling multiple ad slots and multiple ad types on the page. Increasing the flexibility from these features also increases the complexity of the script. In comparison, the existing `run-ad-auction.js` only executes the Protected Audience auction in a single-seller setup and can only handle a single ad slot for a display ad. This PR reintegrates the simplified single-seller Protected-Audience-only auction on the `news/index` page as the simplest demonstration for Protected Audience.

## Affected services

- [ ] Home
- [x] News
- [ ] Shop
- [ ] Travel
- [x] Ad-tech
- [ ] ALL

### Changes to `ad-tech`

* Rename `public/js/ssp/run-ad-auction.js` to `.../run-simple-ad-auction.js`.
* Rename `public/js/ssp/ssp-tag.js` to `.../ad-server-tag.js`.
* Add `HOSTNAME` to JavaScript template variables.

### Changes to `news`

* Rename the current `index.ejs` file to `multiple-ad.ejs`.
* Add a new `index.ejs` with a single ad unit for a fenced-frame display ad in a single-seller auction setup. Add a new view handler for the index page that is executed before the generic view handler. Specify a different `AD_SERVER_TAG_URL` for the index page than that used for the remaining pages. The index page uses `run-simple-ad-auction.js`, whereas the other pages use the more realistic `ssp-tag.js` which can handle multiple ad slots.
* Reformat the `aside` view component which lists the available demo pages on the `news` site.
* Rename `window.PSDemo.getQueryAsArray()` to `getUrlQueryAsArray()`, and `getQueryAsString()` to `getUrlQueryAsString()`; update references accordingly in news pages.

### CODE REVIEW

File: services/news/src/index.ts [hunk 1]
```diff
@@ -33,6 +33,18 @@ app.use(express.static('src/public'));
...
 app.set('views', 'src/views');
 
+app.get('/', async (req: Request, res: Response) => {
+  res.render('index', {
+    TITLE: NEWS_DETAIL,
+    TEXT_LOREM,
+    EXTERNAL_PORT,
+    HOME_HOST,
+    AD_SERVER_TAG_URL: new URL(
+      `https://${AD_SERVER_HOST}:${EXTERNAL_PORT}/js/ssp/run-simple-ad-auction.js`,
```
Comment (siddharth-sahoo) [inline services/news/src/index.ts:43]: Use the SSP service for the simple remarketing use-case to be consistent with the surrounding markdown documentation.

File: services/news/src/index.ts [hunk 2]
```diff
@@ -33,6 +33,18 @@ app.use(express.static('src/public'));
...
+    ).toString(),
+  });
+});
+
 app.get('*', async (req: Request, res: Response) => {
   res.render(req.path.substring(1), {
     TITLE: NEWS_DETAIL,
```
Comment (siddharth-sahoo) [inline services/news/src/index.ts:50]: Rename ssp-tag.js to ad-server-tag.js

File: services/ad-tech/src/public/js/ssp/run-simple-ad-auction.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        status: res.statusText,
+      });
+    }
+  };
+
+  // ****************************************************************
+  // PROTECTED AUDIENCE : RUN AD AUCTION
+  // ****************************************************************
+  if (!navigator.runAdAuction) {
+    return log('stopping becuase Protected Audience is not supported', {});
```
Comment (Seburan) [inline services/ad-tech/src/public/js/ssp/run-simple-ad-auction.js:83]: typo

Comment (siddharth-sahoo) [reply services/ad-tech/src/public/js/ssp/run-simple-ad-auction.js:83]: Good catch!

File: services/news/src/views/multiple-ad.ejs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        href="/img/spy.svg" />
+  <script src="/js/main.js"></script>
+  <script>
+    // Seller acting as ad server is included implictly, and their tag is
+    // included on the page directly. This configuration field should list
+    // additional sellers to include in the ad auctions.
+    const otherSellers = window.PSDemo.getQueryAsArray('otherSellers') || [
+      '<%= SSP_ORIGIN %>',
+      '<%= SSP_A_ORIGIN %>',
+      '<%= SSP_B_ORIGIN %>',
```
Comment (chasinandrew) [inline services/news/src/views/multiple-ad.ejs:23]: Q: To add more sellers to the auction (e.g. SSP-X), would their origin have to be added here? How does the window function with otherSellers work?

**Review by chasinandrew — APPROVED** [review]
Left one general question but LGTM

File: services/news/src/views/multiple-ad.ejs (hunk 4 above)
Comment (siddharth-sahoo) [reply services/news/src/views/multiple-ad.ejs:23]: Renamed to getUrlQueryAsString and getUrlQueryAsArray, both of which are utility functions from https://news/js/main.js.

This snippet of code looks for a URL query called "otherSellers", parses it as an array of comma separated values if found or just defaults to using this list.

Easiest way to include more sellers would be to just add to the list as you mentioned. And a new demo page on the news site could have its own set of default "otherSellers".

**Pull request merged**